marlowe/specification/ReadMe.md

```diff
@@ -0,0 +1,346 @@
+
+
+## Participants in Marlowe Contracts
+
+Marlowe contracts identify each participant by either a *public-key hash (PKH)* or a *role*. Marlowe semantics allow PKHs and roles to be freely mixed in contract logic.
+- A Marlowe PKH corresponds to the hash of a Cardano payment verification key, `Hash PaymentKey`.
+- A Marlowe role corresponds to a Cardano native token with the `TokenName`  for the role. (In Cardano, role names may be zero to thirty-two bytes long.) All of the role tokens in a given Marlowe contract must use the same  `CurrencySymbol` .
+
+
+## Life Cycle of a Marlowe Contract
+
+The execution of a Marlowe contract proceeds as a linear sequence of applications of inputs at the contract's script address.
+1. Role tokens are typically minted prior to or within the creation transaction of the contract, though this is not enforced on-chain.
+2. The creation transaction for the contract stores the state of the contract in the datum at the contract's script address.
+3. Each application of input to the contract updates the state/datum at that same address.
+4. When a contract closes, there is no output to the contract's script address.
+5. If the contract pays funds to a role during application of inputs, those funds are sent to the address of the Marlowe payout-validator script.
+
+Thus each Marlowe contract is a finite linear sequence of continuations at the script address, from creation to closure.
+
+
+## Monetary Policy for Role Tokens
+
+Any Cardano monetary policy may be used to mint the role tokens used in a Marlowe contract. For security in standard Marlowe use cases, a one-time or locked minting policy such as `Plutus.Contracts.Currency.OneShotCurrency` is recommended. Exotic use cases might employ other monetary policies. It is the responsibility both of the developer of the off-chain code managing a contract and also of the user of the contract to verify that the monetary policy of the role tokens meets their security requirements.
+
+
+## Representation of Marlowe Semantics in Plutus
+
+Marlowe semantics are embodied in the Haskell function `computeTransaction` that is generated from [the Isabelle implementation of the Marlowe DSL](https://github.com/input-output-hk/marlowe/tree/master/isabelle/Core). Note that `computeTransaction`  uses the `PlutusTx` types and functions instead of the standard-prelude or GHC-base Haskell ones.
+```haskell
+computeTransaction :: TransactionInput -> State -> Contract -> TransactionOutput
+```
+
+With the exception of `Input`, [the Isabelle types for Marlowe](https://github.com/input-output-hk/marlowe/blob/master/isabelle/Core/SemanticsTypes.thy) directly correspond to their Haskell representations, so below we only provide the definitions that will be later referenced in this document.
```

Comment:
Would it be better to have a stand-alone document that contains all the (Haskell) definitions needed, e.g. in an appendix, rather than having an external dependency like this?

Reply:
I will add an appendix with a correspondence table between Isabelle and Haskell.

Reply:
I'm adding an appendix to this document that will contain sufficient information to make it self contained.